web/src/test/java/org/springframework/security/web/authentication/session/ChangeSessionIdAuthenticationStrategyTests.java

```diff
@@ -40,17 +40,6 @@
 	@Mock
 	private Method method;
 
-	@Test(expected = IllegalStateException.class)
```

Comment:
Another question, now that the reflection operation is removed, what should we test instead of this??

```java 

@Test
		public void applySessionFixation() throws Exception {
			spy(ReflectionUtils.class);
 		Method method = mock(Method.class);
 		MockHttpServletRequest request = new MockHttpServletRequest();
 		request.getSession();
 		when(ReflectionUtils.findMethod(HttpServletRequest.class, "changeSessionId"))
 				.thenReturn(method);
 
 		new ChangeSessionIdAuthenticationStrategy().applySessionFixation(request);
 
 		verifyStatic(ReflectionUtils.class);
 		ReflectionUtils.invokeMethod(same(method), eq(request));
```


I was thinking on something like this, maybe

```java
@Test
	public void applySessionFixation() {
		MockHttpServletRequest request = new MockHttpServletRequest();
		String beforeSessionFixation = request.getSession().getId();

			new ChangeSessionIdAuthenticationStrategy().applySessionFixation(request);

		String afterSessionFixation = request.getSession().getId();

		Assert.assertNotEquals(beforeSessionFixation,afterSessionFixation);
	}
```


what do you think???

Reply:
Another question, now that the reflection operation is removed, what should we test instead of this??

```java 

@Test
		public void applySessionFixation() throws Exception {
			spy(ReflectionUtils.class);
 		Method method = mock(Method.class);
 		MockHttpServletRequest request = new MockHttpServletRequest();
 		request.getSession();
 		when(ReflectionUtils.findMethod(HttpServletRequest.class, "changeSessionId"))
 				.thenReturn(method);
 
 		new ChangeSessionIdAuthenticationStrategy().applySessionFixation(request);
 
 		verifyStatic(ReflectionUtils.class);
 		ReflectionUtils.invokeMethod(same(method), eq(request));
```


I was thinking on something like this, maybe

```java
@Test
	public void applySessionFixation() {
		MockHttpServletRequest request = new MockHttpServletRequest();
		String beforeSessionFixation = request.getSession().getId();

			new ChangeSessionIdAuthenticationStrategy().applySessionFixation(request);

		String afterSessionFixation = request.getSession().getId();

		Assert.assertNotEquals(beforeSessionFixation,afterSessionFixation);
	}
```


what do you think???

Reply:
I like that.

Or you might consider:

```java
@Test
public void whenApplySessionFixationThenChangesIdAndRetainsAttributes() {
    MockHttpServletRequest request = new MockHttpServletRequest();
    request.getSession().setAttribute("custom", "value");

    String id = request.getSession().getId();
    String value = request.getSession().getAttribute("custom");
    
    strategy.applySessionFixation(request);

    assertThat(request.getSession().getId()).isNotEqualTo(id);
    assertThat(request.getSession().getAttribute("custom")).isEqualTo(value);
}
```

Since it's true that the intent of session fixation protection is to change the session id without losing any session attributes.

Reply:
Ok done :)